pkg/sources/storage_manager.go

```diff
@@ -32,7 +35,82 @@ type StorageManager interface {
 	Delete(ctx context.Context, config *config.Config) error
 }
 
-// NewStorageManager creates a new storage manager
-func NewStorageManager() (StorageManager, error) {
-	return nil, fmt.Errorf("storage manager not yet implemented")
+// FileStorageManager implements StorageManager using local filesystem
+type FileStorageManager struct {
+	basePath string
+}
+
+// NewFileStorageManager creates a new file-based storage manager
+func NewFileStorageManager(basePath string) StorageManager {
+	return &FileStorageManager{
+		basePath: basePath,
+	}
+}
+
+// Store saves the registry data to a JSON file
+func (f *FileStorageManager) Store(ctx context.Context, config *config.Config, reg *registry.Registry) error {
+	// Create base directory if it doesn't exist
+	if err := os.MkdirAll(f.basePath, 0755); err != nil {
+		return fmt.Errorf("failed to create storage directory: %w", err)
+	}
+
+	filePath := filepath.Join(f.basePath, RegistryFileName)
+
+	// Marshal registry to JSON with pretty printing for readability
+	data, err := json.MarshalIndent(reg, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal registry data: %w", err)
+	}
+
+	// Write to temporary file first for atomic operation
+	tempPath := filePath + ".tmp"
+	if err := os.WriteFile(tempPath, data, 0644); err != nil {
+		return fmt.Errorf("failed to write temporary registry file: %w", err)
+	}
+
+	// Atomic rename
+	if err := os.Rename(tempPath, filePath); err != nil {
+		// Clean up temp file on error
+		_ = os.Remove(tempPath)
+		return fmt.Errorf("failed to rename registry file: %w", err)
+	}
+
+	return nil
+}
+
+// Get retrieves and parses registry data from the JSON file
+func (f *FileStorageManager) Get(ctx context.Context, config *config.Config) (*registry.Registry, error) {
+	filePath := filepath.Join(f.basePath, RegistryFileName)
+
```

Comment:
There is duplication between FileRegistryDataProvider.GetRegistryData wouldn't it be better to pass a storageManager (or maybe rather a new mini interface) around?

Reply:
Pushed an updated version with common implementation leveraging the storageManager as you said